libraries/AP_HAL_ChibiOS/Semaphores.cpp

```diff
@@ -53,6 +53,9 @@ bool Semaphore::take(uint32_t timeout_ms)
     if (take_nonblocking()) {
         return true;
     }
+#ifdef HAL_BOOTLOADER_BUILD
+    return false;
```

Comment:
why not just use the existing return false at the end? The else is unnecessary

Reply:
because we don't have a hal, so it doesn't compile or link if we use that return